exp/runtime/internal/controllers/extensionconfig_controller_test.go

```diff
@@ -408,6 +409,9 @@ func fakeCASecret(namespace, name string, caData []byte) *corev1.Secret {
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
 			Namespace: namespace,
+			Labels: map[string]string{
```

Comment:
I think this is not necessary anymore as soon as we merge https://github.com/kubernetes-sigs/cluster-api/pull/10633

cc @sbueringer @chrischdi

Reply:
If this was only introduced because otherwise we got not found => yes

Let's definitely revert (after #10633 merges it should work without). This secret *should not* require the cluster-name label (also the secret should not correspond to any cluster)

Reply:
Makes sense! I will wait for it to merge, then rebase and remove this. :slightly_smiling_face:

Reply:
@lentzi90 PR is merged

Reply:
Thanks!
@adilGhaffarDev or @Sunnatillo does anyone of you want to take this over? It may be some time until I can get back to it since I'm on parental leave :innocent:

Reply:
No worries! Enjoy!

Reply:
I can address the comments and push the changes.

Reply:
pushed the changes, please check.